### DISCUSSION TITLE
NextJS Internationalization (i18n) Routing: locale getting undefined

### DISCUSSION THREAD

**Discussion by ucemrecan** [vercel/next.js#53454] [Help] [answered]
### Summary

I'm trying to learn [Next.js Internationalization (i18n) Routing](https://nextjs.org/docs/pages/building-your-application/routing/internationalization) for my personal website. However, the locale, locales, and default locale values are getting as undefined. (I'm using the latest version of Next.js)

### Additional information

```js
Here is my files:

next.config.js:
`/** @type {import('next').NextConfig} */
const nextConfig = {
  i18n: {
    locales: ["tr", "en"],
    defaultLocale: "tr",
  },
};

module.exports = nextConfig;`



locales/tr.js:
export default {
  hello: "Merhaba dünya!",
};


locales/en.js:
export default {
  hello: "Hello world!",
};



app/page.js:
"use client";
import { useRouter } from "next/navigation";

export default function Home() {
  const router = useRouter();
  const { locale, locales, defaultLocale } = router;
  console.log("Locales:", locale, locales, defaultLocale);
  console.log(router);

  return (
    <div>
      <h1>Hello world</h1>
      <p>Current locale: {locale}</p>
      <p>Default locale: {defaultLocale}</p>
      <p>Configured locales: {JSON.stringify(locales)}</p>
    </div>
  );
}


I tried to get information using console.log. The print is as follows:
https://i.stack.imgur.com/SFJKr.png


Lastly, package.json:
{
  "name": "my-app",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "eslint": "8.45.0",
    "eslint-config-next": "13.4.12",
    "next": "13.4.12",
    "react": "18.2.0",
    "react-dom": "18.2.0"
  }
}
I can't find where I made a mistake. Please help me fix it.
```

**Comment by Emircan92** [comment]
Try to import useRouter from 'next/router' 👍

**Reply by Emircan92** [reply]
oh nevermind, this is app router :P

**Comment by caffbit** [comment, marked as answer]
The provided code snippet utilizes the app router to construct a Next.js project. However, the learning material you referred to is based on the pages router documentation. You may need to switch to studying app router materials for your understanding.

For more information on the app router and its usage in Next.js, please visit the link below:
https://nextjs.org/docs/app/building-your-application/routing/internationalization

**Reply by ucemrecan** [reply]
I resolved the issue with the next-intl documentation you shared on the redirected page. I'm grateful, thank you.

https://next-intl-docs.vercel.app/docs/getting-started/app-router-server-components
